snuba/subscriptions/data.py

```diff
@@ -280,6 +332,158 @@ def to_dict(self) -> Mapping[str, Any]:
         return subscription_data_dict
 
 
+@dataclass(frozen=True, kw_only=True)
+class RPCSubscriptionData(_SubscriptionData[TimeSeriesRequest]):
+    """
+    Represents the state of an RPC subscription.
+    """
+
+    time_series_request: str
+
+    request_name: str
+    request_version: str
+
+    def validate(self) -> None:
+        super().validate()
+        if (self.request_name, self.request_version) not in REQUEST_TYPE_ALLOWLIST:
+            raise InvalidSubscriptionError(
+                f"{self.request_name} {self.request_version} not supported."
+            )
+
+        # TODO: Validate no group by, having, order by etc
+
+    def build_request(
+        self,
+        dataset: Dataset,
+        timestamp: datetime,
+        offset: Optional[int],
+        timer: Timer,
+        metrics: Optional[MetricsBackend] = None,
+        referrer: str = SUBSCRIPTION_REFERRER,
+    ) -> TimeSeriesRequest:
+
+        request_class = EndpointTimeSeries().request_class()()
+        request_class.ParseFromString(base64.b64decode(self.time_series_request))
+
+        # TODO: update it to round to the lowest granularity
+        # rounded_ts = int(timestamp.replace(tzinfo=UTC).timestamp() / 15) * 15
+        rounded_ts = (
+            int(timestamp.replace(tzinfo=UTC).timestamp() / self.time_window_sec)
+            * self.time_window_sec
+        )
```

Comment:
hack for now, we want to round this to the lowest available granularity (15s) see https://sentry.slack.com/archives/C07M2RN1AHJ/p1730738626703789